Sources/RouterServiceInterface/PresentationStyle.swift

```diff
@@ -18,14 +19,18 @@ open class PushPresentationStyle: PresentationStyle {
     open func present(
         viewController: UIViewController,
         fromViewController: UIViewController,
-        animated: Bool
+        animated: Bool,
+        completion:(() -> Void)?
     ) {
+        CATransaction.begin()
+        CATransaction.setCompletionBlock(completion)
```

Comment:
What do you think to abstract these `CATransaction` methods and present with completion from **PresentationStyle**?
One suggestion is creating a commitTransaction free method
```swift
func commitTransaction(_ transaction: () -> Void, completion: () -> Void) {
    CATransaction.begin()
    CATransaction.setCompletionBlock(completion)
    transaction()
    CATransaction.commit()
}
```

And another suggestion is to create a UINavigationController extension for `pushViewController(_:)` method. In that way, you can implement future other methods and implementations like `pop(_:)`, `popToRoot(_:)`, etc.

```swift
extension UINavigationController {
    func pushViewController(_ viewController: UIViewController, animated: Bool, completion: @escaping () -> Void) {
        commitTransaction {
            pushViewController(viewController, animated: animated)
        }, completion: completion
    }
}
```

Reply:
I like your suggestion, I am just wondering where to put the free function and the extension method. 
PresentationStyle source file is not the right place. Any suggestion :) ?

Reply:
We can put the free function inside the same file as a private one, as I think we won't need this for somewhere else

Reply:
BTW, we'll also need public extensions for the PresentationStyle to add methods that don't take a completion handler, just so we don't break previous implementations.